src/main/scala/com/atomist/rug/runtime/js/JavaScriptInvokingProjectOperation.scala

```diff
@@ -126,17 +128,36 @@ abstract class JavaScriptInvokingProjectOperation(
 
         p.setValidInputDescription(details.get("validInput").asInstanceOf[String])
         p.describedAs(details.get("description").asInstanceOf[String])
-        details.get("pattern").asInstanceOf[String] match {
+
+        pPattern match {
           case s: String if s.startsWith("@") => DefaultIdentifierResolver.resolve(s.substring(1)) match {
             case Left(_) =>
-              throw new InvalidRugParameterPatternException(s"Unable to recognized predefined validation pattern: $s")
+              throw new InvalidRugParameterPatternException(s"Unable to recognize predefined validation pattern for parameter $pName: $s")
             case Right(pat) => p.setPattern(pat)
           }
-          case s: String if !s.startsWith("^") || !s.endsWith("$") => throw new InvalidRugParameterPatternException(s"Parameter $name does not contain anchors: $s")
+          case s: String if !s.startsWith("^") || !s.endsWith("$") =>
+            throw new InvalidRugParameterPatternException(s"Parameter $pName validation pattern must contain anchors: $s")
           case s: String => p.setPattern(s)
-          case _ =>
+          case _ => throw new InvalidRugParameterPatternException(s"Parameter $pName has no valid validation pattern")
+        }
+
+        val avs = details.get("allowed_values").asInstanceOf[ScriptObjectMirror]
+        val allowedValues: Seq[String] =
```

Comment:
I couldn't think of a way of matching that didn't just rely on conditionals, so I figured simply conditionals were as good an approach as any.